### ISSUE TITLE
uComponents URL picker not working after Umbraco 6.2.4 to 6.2.5 upgrade

### ISSUE THREAD

**Issue by jHodgkinson** [uComponents/uComponents#30]
A 500 server error is thrown when a local Umbraco media file is attempted to be loaded/added.

The following error info below is displayed: 

```
Server Error in '/' Application. Request format is unrecognized for URL unexpectedly ending in '/MediaNodeUrl'.

Description: An unhandled exception occurred during the execution of the current web request.
Please review the stack trace for more information about the error and where it originated in the code.

Exception Details:
System.InvalidOperationException: Request format is unrecognized for URL unexpectedly ending in '/MediaNodeUrl'.

Source Error: An unhandled exception was generated during the execution of the current web request.
Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Stack Trace: [InvalidOperationException: Request format is unrecognized for URL unexpectedly ending in '/MediaNodeUrl'.]
System.Web.Services.Protocols.WebServiceHandlerFactory.CoreGetHandler(Type type, HttpContext context, HttpRequest request, HttpResponse response)
  +668425 System.Web.Services.Protocols.WebServiceHandlerFactory.GetHandler(HttpContext context, String verb, String url, String filePath)
  +306 System.Web.Script.Services.ScriptHandlerFactory.GetHandler(HttpContext context, String requestType, String url, String pathTranslated)
  +89 System.Web.MaterializeHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute()
  +516 System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously)
  +288 Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.0.30319.34209
```

![umberror](https://cloud.githubusercontent.com/assets/5639484/6560819/51171342-c663-11e4-9e95-b89cc59ff878.jpg)

This error has only happened since we have upgraded to Umbraco v6.2.5. If we rollback to Umbraco v6.2.4, then the error goes away (and uComponents works fine).

For reference, here's a topic on the Our Umbraco forum:
https://our.umbraco.org/projects/backoffice-extensions/ucomponents/questionssuggestions/61972-URL-picker-just-spinning,-generating-500-server-error

**Comment by leekelleher** [comment]
It appears that the returning data from `umbraco.library.GetMedia` has changed in v6.2.5.

See this forum post for a code fix:
https://our.umbraco.org/projects/backoffice-extensions/ucomponents/questionssuggestions/61972-URL-picker-just-spinning,-generating-500-server-error?p=0#comment211285

The line in UrlPicker is here:
https://github.com/uComponents/uComponents/blob/master/src/uComponents.DataTypes/UrlPicker/Services/UrlPickerService.cs#L44

**Comment by jHodgkinson** [comment]
thx Lee!  is this the correct link for the latest source for uComponents:

https://github.com/uComponents/uComponents/tree/master/src

**Comment by leekelleher** [comment]
@jHodgkinson Yes, that's the latest stable branch of the source code.

There are some other fixes in the [`dev-6.0.1`](https://github.com/uComponents/uComponents/tree/dev-6.0.1) branch, but I can't guarantee that it is as stable as `master` at this point.

**Comment by jHodgkinson** [comment]
Awesome! Thank you for all your help Lee!


John Hodgkinson
Applications Architect / DBA / Integrations & Development Lead

North Carolina Bar Association
8000 Weston Parkway
Cary, North Carolina 27513
919.657.1579
jHodgkinson@ncbar.org<mailto:jHodgkinson@ncbar.org>
www.ncbar.org<http://www.ncbar.org/>

**Comment by leekelleher** [comment]
I'll get around to doing a new v6.x release soon - _once my current workload eases off a little_ :smirk_cat:

**Comment by darylteo** [comment]
Hi,

I've also encountered this issue. I applied the fix, but still had issues with the javascript.

I'm not sure if the other posters also encountered this issue, but the problem was that the web service call was returning JSON and jQuery immediately picked it up and turned it into an object. This caused an issue with responseText.replace (as an object it doesn't have the replace function).

I fixed this by altering UrlPickerScripts.js. Recompile this together with the changes to UrlPickerService and it should then work.

````
                success: function (responseText) {
                // Strip any <pre> tags and parse the JSON
                var response = jQuery.parseJSON(responseText.replace(/<(\/)?pre[^>]*>/gi, ""));
````

Change to

````
                success:function (response) {
````

Note: my environment has installed the MVC security upgrades (4.0.0.1) as per http://umbraco.com/follow-us/blog-archive/2014/10/16/getting-a-systemwebhttpapicontroller-error which may explain why noone else has encountered this issue.

To accomodate this, perhaps the code should be modified to test for presence of string before trying to replace.

Regards,
Daryl Teo

**Comment by leekelleher** [comment]
@darylteo Thanks for the fixes.  Is there any way we could encourage you to collaborate on a pull-request? (You have the skills :sunglasses:)

**Comment by darylteo** [comment]
Any update?

**Comment by kirkov** [comment]
Changing `InnerXml` to `OuterXml` fixes the issue for me:

var media = XElement.Parse(library.GetMedia((int)id, false).Current.__OuterXml__);

(https://github.com/uComponents/uComponents/blob/85385bfae200c6b831b59720c691654021ac2804/src/uComponents.DataTypes/UrlPicker/Services/UrlPickerService.cs#L44)

**Comment by leekelleher** [comment]
This bug has been fixed in PRs #31 #33 and released in uComponents v6.1.0.
